src/core/src/op/group_query_attention.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/group_query_attention.hpp"
+
+#include "itt.hpp"
+
+using namespace std;
+namespace ov {
+namespace op {
+
+GroupQueryAttention::GroupQueryAttention(const OutputVector& args,
+                                         int64_t num_heads,
+                                         int64_t kv_num_heads,
+                                         float scale,
+                                         bool do_rotary,
+                                         bool rotary_interleaved)
+    : Op(args),
+      m_num_heads(num_heads),
+      m_kv_num_heads(kv_num_heads),
+      m_scale(scale),
+      m_do_rotary(do_rotary),
+      m_rotary_interleaved(rotary_interleaved) {
+    constructor_validate_and_infer_types();
+}
+
+void GroupQueryAttention::validate_and_infer_types() {
+    OV_OP_SCOPE(GroupQueryAttention_validate_and_infer_types);
+    // GQA expectes the following inputs: query, key, value, past_key, past_value, seqlens_k, cos_cache, sin_cache
+    // All qkv's should have the shape [batch, num_heads, seq_len, head_size] ([B, N, S, H])
+    // It has three outputs: output of shape [B, S, N * H], and present_key/value of shape [B, N, S, H]
+    // seqlens_k is number of 1's in the attention_mask minus 1
+
+    PartialShape q_shape = get_input_partial_shape(0);
+    NODE_VALIDATION_CHECK(this, q_shape[3].is_static(), "GroupQueryAttention: head size should not be dynamic");
```

Comment:
The q_shape cannot be dynamic and must have correct rank to access [3] and 

Why dimension [3] cannot be dynamic?